src/transactions/transactions.service.ts

```diff
@@ -0,0 +1,213 @@
+import { Card } from 'src/cards/card.model';
```

Comment:
Consider adding validation or using a validation library like class-validator to ensure the input is valid.